pkg/controller/controller.go

```diff
@@ -370,29 +370,25 @@ func (c *Controller) getDesiredWorkers(
 		return convertDesiredReplicasWithRules(desiredWorkers, minWorkers, maxWorkers)
 	}
 
-	if queueMessages > 0 {
+	if queueMessages > 0 && idleWorkers <= 0 {
 		// return the current replicas if the change would be too small
 		if math.Abs(1.0-usageRatio) <= tolerance {
 			return currentWorkers
 		}
-
 		if queueMessages < targetMessagesPerWorker {
 			return currentWorkers
 		}
 
-		desiredWorkers := int32(math.Ceil(usageRatio * float64(currentWorkers)))
+		desiredWorkers := int32(math.Ceil(usageRatio))
 		// to prevent scaling down of workers which could be doing processing
-		if desiredWorkers < currentWorkers {
-			return currentWorkers
-		}
+
```

Comment:
Why are we changing the formula here.. Please suggest. We should not be ideally changing these code in this PR. Please raise a separate PR with the reason mentioned.